### ISSUE TITLE
[Feature]: Show "No Data Available" message on table widget when there is no data

### ISSUE THREAD

**Issue by danciaclara** [appsmithorg/appsmith#12709]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

It will be useful to have a message show up on the table like "No Data Available" or something along those lines when there is no data to display on the table. Today it shows a blank table when you empty the table data.

![Uploading Screenshot 2022-04-08 at 2.30.51 PM.png…]()

### Why should this be worked on?

A blank table can be confusing to users so having a clear message will be a good user experience.

Was also asked by a discord user
https://discord.com/channels/725602949748752515/760761686549463060/961903375933407232

**Comment by dilippitchika** [comment]
Assuming 10% of our table users see this
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 500 |
| Effort (months)  | 0.25 |

**Comment by dilippitchika** [comment]
@sbalaji1192 can you take a look and let me know if we can add this? If yes, I will ask someone else to pick this up.

**Comment by sbalaji1192** [comment]
@dilippitchika We can do this. Currently, we intentionally fill the table with empty cells when the tableData is empty.

**Comment by dilippitchika** [comment]
@sbalaji1192 do you know why we do this? Any catch with implementing this?

**Comment by dilippitchika** [comment]
Ok, @somangshu or @Nikhil-Nandagopal may know. Folks any reason we took this behavior?

@sbalaji1192 will this show up when the evaluations are still going on? If we add this will it cause issues in #12308?

**Comment by somangshu** [comment]
I feel it is expected to see empty columns in the table, Atleast the headers, It gives the end user an idea of the data schema. 

@dilippitchika Maybe we leave the column heading as it is and show this only in place of the data rows?

**Comment by dilippitchika** [comment]
@somangshu you are right we can show the schema for the table. I thought this already exists in the table widget. 

This will also be useful when data is filtered or searched but no results are found and other cases. I will add a small PRD and get this designed.

**Comment by vasanth-appsmith** [comment]
@dilippitchika  Design is completed. Here is the  [Figma Link](https://www.figma.com/file/O113mgBknwDa1cLl7zx3u0/Table-Widget-with-no-data?node-id=12%3A6083)

**Comment by dilippitchika** [comment]
We need to do a POC here

**Comment by dilippitchika** [comment]
@dilippitchika to check this again

**Comment by Naik1994** [comment]
Hi Team,

Is this feature implemented in any release ?

**Comment by somangshu** [comment]
Hey @Naik1994, this is not a priority for us right now, we dont expect to pick this up anytime soon. 

If you want a similar behaviour, a workaround is to use the tabs widget, you can use a JS expression in the default tab based on the data that you load on the table and show an empty state. 

You also have an option on tabs widget to hide the tab to make it look neater.

**Comment by leonlazic** [comment]
Just adding that I would like this feature too. I would make it much cleaner for the end user.

**Comment by carinanfonseca** [comment]
thank you for your feedback @leonlazic! we'll consider that.

**Comment by ecodylan** [comment]
Also agree that it would be valuable to have this feature! Even the ability to choose between displaying a no data message or displaying the headers with empty cells.
